packages/php-wasm/compile/build.js

```diff
@@ -167,7 +167,10 @@ if (!requestedVersion || requestedVersion === 'undefined') {
 const sourceDir = path.dirname(new URL(import.meta.url).pathname);
 
 // Build the base image
-await asyncSpawn('make', ['base-image'], { cwd: sourceDir, stdio: 'inherit' });
+await asyncSpawn('make', ['all', 'base-image'], {
```

Comment:
does the `base-image` depend on the `all` target or is that incidental here?

otherwise it seems like the `Makefile` would be the place to list dependencies, wouldn't it? why aren't we putting it there?

Reply:
@dmsnell The `base-image` target does not depend on the `all` target. (Its generally poor practice to use a `PHONY` target as a dependency because that means anything downstream from that target would be rebuilt on every invocation.)

https://github.com/WordPress/wordpress-playground/blob/db34c273d2810b5f3d0795fd39ce69e06f0bc17f/packages/php-wasm/compile/Makefile#L1-L2

Reply:
@dmsnell Actually its fine so long as phony targets are used as depencencies of other phony targets:

https://github.com/WordPress/wordpress-playground/blob/c4cfd1e1b05f25e4eae31b91393d9037fa50bc6e/packages/php-wasm/compile/Makefile#L9-L13

Reply:
@seanmorris I think we might have jumped a little too quickly here. I wasn't asking if `base-image` currently is encoded as depending on `all`, but rather was asking about your change, which seemed to be encoding a dependency _outside_ of the `Makefile`. previously we asked for `base-image`, but with your change we were also asking for `all`. this communicated to me that `all` is necessary to build `base-image`, and if that were actually true then I think it would be best to leave the dependencies inside `make` where they are handled robustly.

as for `.PHONY` I think there may also be some misunderstandings. we should use them wherever they are part of a dependency hierarchy, whether or not they are a dependency of another `.PHONY` or not. the distinction is purely whether the `.PHONY` targets need to be rebuilt every time `make` runs. for example, it could be that some build target needs to download a configuration from the internet. that download would be a candidate `.PHONY` target, but that doesn't force other targets to rebuild because we could make those other targets depend on the downloaded configuration file directly, and simply not update it if the download is the same as it previously was.

```makefile
.PHONY: fetch-config
fetch-config:
	curl https://my.site/config.json -o /tmp/config.json
	diff -q -s /tmp/config.json ./config.json
	if [ $? -ne 0 ]; then mv /tmp/config.json ./config.json; fi

libsomething: ./config.json
	build-lib-something
```

making `all` a `.PHONY` target is common because people generally don't want to put sub-dependencies on the `all` target, and even if `all` is `.PHONY` and runs every time, if the dependencies of the dependencies haven't changed than nothing much will run.

so we want to be judicious and thoughtful when applying `.PHONY`, but it's also not something we want to try and avoid; we only want to understand its purpose and value and use it where appropriate.

---

I can see that we have a number of ways we could improve the `Makefile`, but those don't need to occur in this PR.